shotover/src/transforms/kafka/sink_cluster/mod.rs

```diff
@@ -2613,150 +2628,166 @@ impl KafkaSinkCluster {
+        let up_shotover_nodes: Vec<_> = self
             .shotover_nodes
             .iter()
-            .map(|shotover_node| {
-                (
-                    shotover_node.broker_id,
-                    MetadataResponseBroker::default()
-                        .with_host(shotover_node.address.host.clone())
-                        .with_port(shotover_node.address.port)
-                        .with_rack(Some(shotover_node.rack.clone())),
-                )
-            })
+            .filter(|shotover_node| shotover_node.is_up())
             .collect();
 
-        // Overwrite the list of partitions to point at all shotover nodes within the same rack
-        for (_, topic) in &mut metadata.topics {
-            for partition in &mut topic.partitions {
-                // Deterministically choose a single shotover node in the rack as leader based on topic + partition id
-                if let Some(leader_rack) = self
-                    .nodes
-                    .iter()
-                    .find(|x| x.broker_id == *partition.leader_id)
-                    .map(|x| x.rack.clone())
-                {
-                    let shotover_nodes_in_rack: Vec<_> = self
-                        .shotover_nodes
-                        .iter()
-                        .filter(|shotover_node| {
-                            leader_rack
-                                .as_ref()
-                                .map(|rack| rack == &shotover_node.rack)
-                                .unwrap_or(true)
-                        })
-                        .collect();
-                    let hash = hash_partition(topic.topic_id, partition.partition_index);
-                    let shotover_node =
-                        &shotover_nodes_in_rack[hash % shotover_nodes_in_rack.len()];
-                    partition.leader_id = shotover_node.broker_id;
-                } else {
-                    // If the cluster detects the broker is down it will not be included in the metadata list of brokers.
-                    // In that case we wont find a broker and we should just mark the leader as not existing by setting it to -1
-                    // The brokers also set the leader to -1 if they detect it is down, so this matches their behaviour.
-                    partition.leader_id = BrokerId(-1);
-                }
-
-                // Every replica node has its entire corresponding shotover rack included.
-                // Since we can set as many replica nodes as we like, we take this all out approach.
-                // This ensures that:
-                // * metadata is deterministic and therefore the same on all shotover nodes
-                // * clients evenly distribute their queries across shotover nodes
-                let mut shotover_replica_nodes = vec![];
-                for replica_node in &partition.replica_nodes {
-                    if let Some(rack) = self
+        if !up_shotover_nodes.is_empty() {
```

Comment:
since the local shotover node will never be down this will always be true.

Reply:
removed this if condition and added a comment about it